syscalls_darwin.go

```diff
@@ -159,7 +159,16 @@ func (t *tunReadCloser) Write(from []byte) (int, error) {
 	}
 	t.wBuf = t.wBuf[:len(from)+4]
 
-	t.wBuf[3] = 2 // Family: IP (2)
+	// Determine the IP Family for the NULL L2 Header
+	ipVer := from[0] >> 4
```

Comment:
My thought would be to put a check at the top of the function and if `from` is 0 length or nil, return `(0, nil)` back. Writing a malformed null header in that scenario seems like bad behavior to me. Thoughts?